src/ucs/stats/libstats.h

```diff
@@ -56,14 +57,29 @@ struct ucs_stats_class {
 
 /* In-memory statistics node */
 struct ucs_stats_node {
-    ucs_stats_class_t    *cls;
-    ucs_stats_node_t     *parent;
-    char                 name[UCS_STAT_NAME_MAX + 1];
-    ucs_list_link_t      list;
-    ucs_list_link_t      children[UCS_STATS_CHILDREN_LAST];
-    ucs_stats_counter_t  counters[];
+    ucs_stats_class_t        *cls;
+    ucs_stats_node_t         *parent;
+    char                     name[UCS_STAT_NAME_MAX + 1];
+    ucs_list_link_t          list;
+    ucs_list_link_t          children[UCS_STATS_CHILDREN_LAST];
+    ucs_stats_counter_list_t **counters_sum;
```

Comment:
instead of keeping back reference to know which coulters should cleaned, i think it would be better to keep list of nodes which should be aggregated (for each class), and just remove the node from that list when it's released. also, need to keep a bitmap of enabled counters for each node, according to the filter. this would also prepare for next part when need to add generic statistic filter.